### PR TITLE
tcs_startup.sh: Add kv_storage startup

### PR DESCRIPTION
**Pull request by danintel**
tcs_startup.sh: Add kv_storage startup

tcs_startup.sh
- Start kv_storage before everything else:
   `kv_storage --bind http://avalon-lmdb:9090`
   (or the value provided by the `-l` option)
- Add `-s` option to optionally not start kv_storage

Dockerfiles
- Add `-s` option when caling tcs_startup.sh from Docker

Signed-off-by: danintel <daniel.anderson@intel.com>

### CODE REVIEW

File: scripts/tcs_startup.sh [hunk 1]
```diff
@@ -14,30 +14,41 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+KV_STORAGE="$TCF_HOME/tools/build/_dev/bin/kv_storage"
 enclave_manager="${TCF_HOME}/examples/enclave_manager/tcf_enclave_manager/enclave_manager.py"
 listener="${TCF_HOME}/common/python/connectors/direct/tcs_listener/tcs_listener.py"
 version="$(cat ${TCF_HOME}/VERSION)"
+# Default values
+DEFAULT_LISTENER_PORT=1947 # set in config/tcs_config.toml
+START_KV_STORAGE=1 # default if -s not passed
```
Comment (Ram-srini) [inline scripts/tcs_startup.sh]: Where is "-s" parameter is checked in the script? By default START_KV_STORAGE is set 1 and it is starting kv_storage service in tcf docker container as well.

Comment (manojgop) [reply scripts/tcs_startup.sh]: Looks fine to me. "-s" option is passed with tcs_startup.sh in docker compose file. So KV storage won't start.

Comment (manojgop) [reply scripts/tcs_startup.sh]: Another option is to remove the "-s" option from this script and start kv storage separately in stand alone build

File: ci/tcf-tests.yaml [hunk 2]
```diff
@@ -25,7 +25,7 @@ services:
         - no_proxy
     working_dir: "/project/TrustedComputeFramework/"
     entrypoint: "bash -c \"\
-        source scripts/tcs_startup.sh -y \
+        source scripts/tcs_startup.sh -y -s -l http://avalon-lmdb:9090 \
```
Comment (pankajgoyal2) [inline ci/tcf-tests.yaml]: Why don't you remove running kv_storage service from tcs_startup.sh at all. Now, this is a seperate service, be it standalone or docker mode, we should start this service seperately.

File: scripts/tcs_startup.sh (hunk 1 above)
Comment (danintel) [reply scripts/tcs_startup.sh]: @Ram-srini: The `-s` option is parsed by `getopts` around line 75.
The `-s` option is passed from docker com a Dockerfile as mentioned by @manojgop.

@manojgop Yes, `kv_storage` can be started separately and that will work. But the purpose of `tcs_startup.sh` is to simplify things and start all the components.

File: ci/tcf-tests.yaml (hunk 2 above)
Comment (danintel) [reply ci/tcf-tests.yaml]: @Ram-srini: consider both PR #214 and #217. PR #217 builds on the changes in PR #214 These PRs are urgently needed to fix the regressions in Avalon so I can deliver a working version of Avalon that is needed by a partner who wants to use Avalon.

@pankajgoyal2 `tcs_startup.sh` starts all the services needed by Avalon in one simple startup script, including the enclave manager and listener. Avalon is very complex and there is no need to make it more difficult to use.

Comment (danintel) [reply ci/tcf-tests.yaml]: @Ram-srini: to reduce confusion I removed the duplicate changes from PR #214.  Please consider both PR #214 and this PR separately.  I will merge after approval.

File: scripts/tcs_startup.sh (hunk 1 above)
Comment (Ram-srini) [reply scripts/tcs_startup.sh]: Got it. Thanks.

**Comment by danintel** [comment]
I'm closing this PR and opening a new PR #223 .
PR #223 is different in that `kv_storage` is started only if `tcs_startup.sh -s` is used, so it should be less controversial.

**Pull request closed without merging**